File: serene/src/Serene.Web/Modules/Administration/Language/LanguageColumns.cs
namespace Serene.Administration.Forms;

[ColumnsScript("Administration.Language")]
[BasedOnRow(typeof(LanguageRow), CheckNames = true)]
public class LanguageColumns
{
    [EditLink]
    public string LanguageId { get; set; }
    [EditLink, SortOrder(1)]
    public string LanguageName { get; set; }
}

File: serene/src/Serene.Web/Modules/ServerTypes/Administration/LanguageColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LanguageRow } from "./LanguageRow";

export interface LanguageColumns {
    LanguageId: Column<LanguageRow>;
    LanguageName: Column<LanguageRow>;
}

export class LanguageColumns extends ColumnsBase<LanguageRow> {
    static readonly columnsKey = 'Administration.Language';
    static readonly Fields = fieldsProxy<LanguageColumns>();
}